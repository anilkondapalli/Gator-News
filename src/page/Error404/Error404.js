import React from "react";
import { Link } from "react-router-dom";
import Error404Image from "../../assets/png/404-Error.png";
import Logo from "../../assets/png/gator-logo.png";
import "./Error404.scss";

export default function Error404() {

    return(
        <div className="error404">
        <img src={Logo} alt="GatorLogo" />
        <img src={Error404Image} alt="Error404" />
        <Link to="/">Back to top</Link>
      </div>
    )
}