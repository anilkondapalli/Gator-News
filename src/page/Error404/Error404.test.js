
import React from "react";
import Error404 from './Error404';
import { BrowserRouter as Router } from 'react-router-dom'; 
import Error404Image from "../../assets/png/404-Error.png";
import { Link } from 'react-router-dom';
import { StaticRouter } from 'react-router'
import renderer from 'react-test-renderer';

// Since Link cant be used outside router we have to rap all the calls with <Router>
describe('Error404', () => {
    test('Render Error404 Page', () => {
        <Router>
            render(<Error404/>);
        </Router> 
    });

    test('Error404 must have src={Logo} and alt="GatorLogo"', () => {
        <Router>
            render(<Error404/>);
            const logo = screen.getByRole('img');
            expect(logo).toHaveAttribute('src', Logo);
            expect(logo).toHaveAttribute('alt', 'GatorLogo');
        </Router>
      });

      test('Error404 must have src={Error404Image} and alt="Error404', () => {
        <Router>
            render(<Error404/>);
            const logo = screen.getByRole('img');
            expect(logo).toHaveAttribute('src', Error404Image);
            expect(logo).toHaveAttribute('alt', 'Error404');
        </Router>
      });

      test('Link in Error 404 matches snapshot', () => {
        const error404Page = renderer.create(
          <StaticRouter>
            <Link to="/" />
          </StaticRouter>
        );
      
        let componentTree = error404Page.toJSON();
        expect(componentTree).toMatchSnapshot();
      });
});