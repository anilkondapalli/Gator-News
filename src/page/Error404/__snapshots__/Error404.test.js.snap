// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 Link in Error 404 matches snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;
